agreement/agreement.go

```diff
@@ -683,8 +683,9 @@ func (w *AgreementWorker) syncOnInit() error {
 					}
 				}
 				glog.V(3).Infof(logString(fmt.Sprintf("added agreement %v to policy agreement counter.", ag.CurrentAgreementId)))
+			} else if ag.AgreementTimeout == 0 {
```

Comment:
Normally, every agreement will fall into the previous else if at line 665. I think your new code needs to be a standalone if.  Maybe do it at the top of the loop.

Reply:
Moved that piece to it's own if